### PR TITLE
feat: dashboard lead time gauge

### PR DESCRIPTION
**Pull request by Tymek**
image

### CODE REVIEW

File: frontend/src/component/executiveDashboard/Gauge/Gauge.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+const degreesToRadians = (degrees: number) => degrees * (Math.PI / 180);
+
+const normalizeValue = (value: number, min: number, max: number) =>
+    (value - min) / (max - min);
+
+const describeArc = (radius: number, startAngle: number, endAngle: number) => {
+    const start = polarToCartesian(0, 0, radius, endAngle);
+    const end = polarToCartesian(0, 0, radius, startAngle);
+    const largeArcFlag = endAngle - startAngle <= Math.PI ? '0' : '1';
+    const d = [
```
Comment (FredrikOseberg) [inline frontend/src/component/executiveDashboard/Gauge/Gauge.tsx:26]: This variable name is very obscure. Can we improve the naming?

File: frontend/src/component/executiveDashboard/Gauge/Gauge.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    const angles = [180 + 50, 180 + 90 + 40];
+
+    return (
+        <>
+            {angles.map(degreesToRadians).map((angle) => {
+                const start = polarToCartesian(0, 0, startRadius, angle);
+                const end = polarToCartesian(0, 0, endRadius, angle);
+
+                return (
+                    <>
```
Comment (FredrikOseberg) [inline frontend/src/component/executiveDashboard/Gauge/Gauge.tsx:58]: Don't we need to encapsulate these with svg?

EDIT: I see they are encapsulated further down. But maybe a comment that this needs to be used in the context of an svg tag.

File: frontend/src/themes/dark-theme.ts [hunk 3]
```diff
@@ -278,6 +278,19 @@ const theme = {
...
+        /**
+         * Dashboard and charts
+         */
+        charts: {
+            gauge: {
+                gradientStart: '#4C4992',
+                gradientEnd: '#9792ED',
+                background: '#39384C',
+                sectionLine: '#8c89bf',
+                text: colors.grey[800],
```
Comment (FredrikOseberg) [inline frontend/src/themes/dark-theme.ts:291]: These should be coming from the theme? Or align with Nicolae which theme colors to use instead.

**Review by FredrikOseberg — APPROVED** [review]
Approved with some comments. This seems to be a very complex component, I wish there was an easier way to do this as it will add a lot of cognitive load to future work in this area. But, I trust that you use your judgment to find the simplest solution and it may be that the graphics of the widget require this approach.

File: frontend/src/themes/dark-theme.ts (hunk 3 above)
Comment (Tymek) [reply frontend/src/themes/dark-theme.ts:291]: That's the approach in entire dark theme, unfortunately

Comment (Tymek) [reply frontend/src/themes/dark-theme.ts:291]: I aligned with Nicolae. We already have some of the colors used in theme, but in different context, and I didn't want to mix it. Using `palette.secondary.main` next to `palette.spotlight.border` is confusing. I don't want other components to influence colors on this chart.

Comment (FredrikOseberg) [reply frontend/src/themes/dark-theme.ts:291]: We don't want hard coded colors. Period. So we need to align and add these colors to theme as more appropriate labels. This is technical debt accumulation.

**Pull request merged**